File: data/data_transpiled/at_coder/arc017/A/3858876.py
def import sys
class Main ( object ) :
    def __init__ ( self ) :
        self.Scanner = sys.stdin
        self.N = self.Scanner.__next__ ( )
        for i in range ( 2 , self.N * i + 1 ) :
            if self.N % i == 0 :
                self.Print ( "NO" )
                return
        self.Print ( "YES" )

File: data/data_original/at_coder/arc017/A/3858876.java
import java.util.*;

public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int N = sc.nextInt();
    for (int i = 2; i * i <= N; i++) {
      if (N % i == 0) {
        System.out.println("NO");
        return;
      }
    }
    System.out.println("YES");
  }
}